### PR TITLE
feat: support using Lambda layers

### PR DESCRIPTION
**Pull request by theburningmonk**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -48,6 +72,49 @@ class LumigoPlugin {
 		await this.wrapFunctions();
 	}
 
+	async getLatestNodeLayerVersionArn(layerArn) {
+		const resp = await http.get(nodeLayerVersionsUrl);
+		const pattern = `${layerArn}:(\\d*)`;
```
Comment (efimk-lu) [inline src/index.js:77]: Why `*` and not `+`
Any special reason for using grouping?

File: src/index.js [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
+	"https://raw.githubusercontent.com/lumigo-io/lumigo-node/master/layers/LAYERS12x.md";
+const pythonLayerVersionsUrl =
+	"https://raw.githubusercontent.com/lumigo-io/python_tracer/master/layers/LAYERS37.md";
+
 const NodePackageManagers = {
 	NPM: "npm",
 	Yarn: "yarn"
 };
 
+const LayerArns = {
```
Comment (efimk-lu) [inline src/index.js:18]: Why not put it inside the class?

Comment (efimk-lu) [reply src/index.js:18]: It's an internal class implementation

File: src/index.js [hunk 3]
```diff
@@ -62,79 +129,108 @@ class LumigoPlugin {
...
-					func,
-					token,
-					parameters
-				);
+				const funcRuntime = func.runtime || runtime;
+				func.layers = func.layers || [];
+				const layer = await this.getLayerArn(funcRuntime);
+				func.layers.push(layer);
+				func.environment = func.environment || {};
+				func.environment["LUMIGO_ORIGINAL_HANDLER"] = func.handler;
```
Comment (efimk-lu) [inline src/index.js:145]: I'm missing here also the token environment variable

File: src/index.test.js [hunk 4]
```diff
@@ -298,6 +299,82 @@ describe("Lumigo plugin (node.js)", () => {
 				);
 			});
 		});
+
+		describe("when useLayers is true", () => {
```
Comment (efimk-lu) [inline src/index.test.js:303]: I'm missing also an integration test, see https://github.com/lumigo-io/serverless-lumigo-plugin/blob/8668fa2c6a62087cbda8108369112d9bbcb42709/scripts/checks.sh#L28

File: src/index.js (hunk 1 above)
Comment (theburningmonk) [reply src/index.js:77]: good shout

File: src/index.js (hunk 2 above)
Comment (theburningmonk) [reply src/index.js:18]: don't need to, it's just a dictionary

File: src/index.js (hunk 3 above)
Comment (theburningmonk) [reply src/index.js:145]: good catch

**Comment by theburningmonk** [comment]
@efimk-lu I made those couple of changes you suggested, and ran the check script locally

**Review by efimk-lu — APPROVED** [review]

**Pull request merged**